ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/KeccakHasher.java

```diff
@@ -38,14 +38,13 @@ public PoWSolution hash(
       final long number,
       final EpochCalculator epochCalc,
       final byte[] prePowHash) {
-    Bytes input = Bytes.wrap(Bytes.wrap(prePowHash), Bytes.ofUnsignedLong(nonce));
 
     MessageDigest digest = KECCAK_256.get();
-    digest.update(input.toArrayUnsafe());
-    byte[] result = digest.digest();
-    Bytes32 resultBytes = Bytes32.wrap(result);
+    digest.update(prePowHash);
+    digest.update(Bytes.ofUnsignedLong(nonce).toArrayUnsafe());
+    Bytes32 solution = Bytes32.wrap(digest.digest());
+    Hash mixHash = Hash.wrap(solution);
```

Comment:
Much clearer now. I would suggest just leaving a block comment here as well, that lays out exactly how the mixhash works in our case, and why it's different than the default Ethash - just for future development reference if others want to experiment with other hashing algos beyond just Keccak256: 

```
Block is valid if keccak256(keccak256(rlp(unsealed header)), nonce) <= 2^256 / difficulty
mixhash = keccak256(rlp(unsealed header)) which is stored in the block. Currently this process is ethash(rlp(unsealed header))
So to validate a block we check keccak256(mixhash, nonce) <= 2^256 / difficulty
```

Reply:
Added the comment to the hasher class.